### PR TITLE
store: sets flagPreviousPresumeKNE after when a stmt finishes which sets Pre…

### PR DESCRIPTION
**Pull request by ekexium**
Signed-off-by: ekexium <eke@fastmail.com>

### What problem does this PR solve?

Issue Number: ref #36579 

Problem Summary:

Fix a bug caused by the loss of PresumeKNE flag in statement retry in pessimistic transactions.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: DEPS.bzl [hunk 1]
```diff
@@ -3401,8 +3401,9 @@ def go_deps():
         name = "com_github_tikv_client_go_v2",
         build_file_proto_mode = "disable_global",
         importpath = "github.com/tikv/client-go/v2",
-        sum = "h1:pZoPlKWCecxJKL8oRq/se71RTljYDrQlZQ2NzKkMYi0=",
-        version = "v2.0.1-0.20220830073839-0130f767386c",
+        replace = "github.com/ekexium/client-go/v2",
```
Comment (sticnarf) [inline DEPS.bzl:3404]: Don't forget to re-generate bazel files.

Comment (ekexium) [reply DEPS.bzl:3404]: Always 😢

**Review by sticnarf — APPROVED** [review]
LGTM

**Review by cfzjywxk — APPROVED** [review]

**Pull request merged**